### PR TITLE
Tasks/ecer 1764 recaptcha

### PR DESCRIPTION
**Pull request by SoLetsDev**
---
name: Pull Request Template
about: Template for creating pull requests
---

## Title
Parent
ECER-1596: Implement recaptcha
subtasks ECER-1765 - create FE captcha component and connect it to the backend. 

## Description

- Registered for recaptcha with Google. 
- Implemented BE logic to make a callout to verify recpatcha token. 
- created backend endpoint to return siteKey for the frontend to use with reCaptcha.

## Various refactor

- changed URLs to be camelCased instead of hyphenated. Ex. character-reference -> characterReference. 
- for References endpoint. Returning a ProblemDetail instead of string for errors. Utilized details field for the frontend to display more specific messages for errors.  
- Small bugfix for un-initialized variable.

## Checklist
- [x] I have tested these changes locally.
- [x] Changes to backend endpoints are covered by passing integration tests.
- [ ] I have added or updated the necessary documentation.

## Screenshots (if applicable)
image

image

### CODE REVIEW

**Comment by farzadnadiri** [comment]
The backend piece seems ok to me. Nice work!

**Comment by farzadnadiri** [comment]
The backend piece seems ok to me. Nice work!

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceRecaptcha.vue [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      type: Object as () => EceRecaptchaProps,
+      required: true,
+    },
+  },
+  emits: {
+    "update:model-value": (_recaptchaToken: string) => true,
+  },
+  async setup() {
+    const siteKey = await getRecaptchaSiteKey();
+    useScriptTag("https://www.recaptcha.net/recaptcha/api.js?onload=recaptchaOnloadCallback&render=explicit", () => {}, { async: true, defer: true });
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceRecaptcha.vue:36]: Cool, never seen this hook

**Review by pcatkins — APPROVED** [review]
FE looks good!

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceRecaptcha.vue (hunk 1 above)
Comment (SoLetsDev) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceRecaptcha.vue:36]: Yeah it's pretty interesting, loads scripts only when you use the component.

**Pull request merged**